lib/components/team/InviteLink.js

```diff
@@ -19,20 +19,27 @@ class InviteLink extends React.Component {
 
   async fetchComponentData() {
     const inviteLink = await apiRequest(null, 'get', `/teams/${this.props.team}/invites/generate`)
-    return inviteLink
+    const state = copy(this.state)
+    if (typeof inviteLink !== 'string') {
+      state.dataError = true
+    }
+    state.inviteLink = inviteLink
+    state.dataLoading = false
+    this.setState(state)
   }
 
   componentDidMount() {
-    return this.fetchComponentData()
-      .then(inviteLink => {
-        const state = copy(this.state)
-        if (typeof inviteLink !== 'string') {
-          state.dataError = true
-        }
-        state.inviteLink = inviteLink
-        state.dataLoading = false
-        this.setState(state)
-      })
+    this.fetchComponentData()
```

Comment:
should this have an await?

Reply:
No, as the function does not need to wait for the promise to finish.

Also, you cannot `await` here due to being unable to make `componentDidMount` an `async` function, instead, we would need to return the promise and use `.then`

Reply:
as discussed, happy to approve